db/error_handler.cc

```diff
@@ -416,6 +429,13 @@ const Status& ErrorHandler::SetBGError(const IOStatus& bg_io_err,
     EventHelpers::NotifyOnBackgroundError(db_options_.listeners, reason,
                                           &new_bg_io_err, db_mutex_,
                                           &auto_recovery);
+    if (bg_error_stats_ != nullptr) {
+      RecordTick(bg_error_stats_.get(), ERROR_HANDLER_BG_ERROR_COUNT);
+      RecordTick(bg_error_stats_.get(), ERROR_HANDLER_BG_IO_ERROR_COUNT);
+      RecordTick(bg_error_stats_.get(),
```

Comment:
Could we break this into 2 stats to distinguish between retryable errors RocksDB attempted to recover from and retryable errors that it didn't attempt? It'll help indirectly track the WAL related retryable errors.

Reply:
I will add a new statistics call ERROR_HANDLER_BG_RETRYABLE_IO_ERROR_RECOVER_ATTEMPTED.

Reply:
ERROR_HANDLER_BG_IO_ERROR_COUNT - ERROR_HANDLER_AUTORESUME_COUNT is the retryable errors that we do not attempt the recover.